tests/jax/components/building/adders_test.py

```diff
@@ -15,135 +15,309 @@
+    """Creates a uniform adder priority fixture
+
+    Returns:
+        UniformAdderPriority.
+    """
+
+    priority = UniformAdderPriority()
+    return priority
 
 
 @pytest.fixture
-def test_system_parallel_sequence_adder() -> System:
-    """Dummy system with zero components."""
-    return TestSystemWithParallelSequenceAdder()
+def mock_env_specs() -> MAEnvironmentSpec:
+    """Creates a mock environment spec
+
+    Returns:
+        MAEnvironmentSpec.
+    """
+
+    return make_fake_env_specs()
+
+
+def test_parallel_sequence_adder(
+    mock_builder: Builder,
+    parallel_sequence_adder: ParallelSequenceAdder,
+) -> None:
+    """Test sequence adder callbacks.
+
+    Args:
+        mock_builder: Fixture SystemBuilder.
+        parallel_sequence_adder: Fixture ParallelSequenceAdder.
+
+    Returns:
+        None
+    """
+    # ParallelSequenceAdderConfig parameters set correctly
+    assert parallel_sequence_adder.config.sequence_length == 1
+    assert parallel_sequence_adder.config.period == 1
+    assert parallel_sequence_adder.config.use_next_extras is True
 
+    parallel_sequence_adder.on_building_init_start(builder=mock_builder)
+    assert (
+        mock_builder.store.sequence_length
+        == parallel_sequence_adder.config.sequence_length
+    )
+
+    parallel_sequence_adder.on_building_executor_adder(builder=mock_builder)
+    assert type(mock_builder.store.adder) == reverb_adders.ParallelSequenceAdder
+
+    # Reverb ParallelSequenceAdder args have been set correctly
+    assert (
+        mock_builder.store.adder._sequence_length
+        == parallel_sequence_adder.config.sequence_length
+    )
+    assert mock_builder.store.adder._period == parallel_sequence_adder.config.period
+    assert (
+        mock_builder.store.adder._use_next_extras
+        == parallel_sequence_adder.config.use_next_extras
+    )
+    assert mock_builder.store.adder._client == mock_builder.store.data_server_client
+    assert mock_builder.store.adder._priority_fns == mock_builder.store.priority_fns
+    assert (
+        mock_builder.store.adder._net_ids_to_keys == mock_builder.store.unique_net_keys
+    )
+    assert (
+        mock_builder.store.adder._table_network_config
+        == mock_builder.store.table_network_config
+    )
+    assert parallel_sequence_adder.name() == "executor_adder"
+
+
+def test_parallel_sequence_adder_signature(
+    mock_builder: Builder,
+    parallel_sequence_adder: ParallelSequenceAdder,
+    parallel_sequence_adder_signature: ParallelSequenceAdderSignature,
+    mock_env_specs: MAEnvironmentSpec,
+) -> None:
+    """Test sequence adder signature callback.
+
+    Args:
+        mock_builder: Fixture SystemBuilder.
+        parallel_sequence_adder_signature: Fixture ParallelSequenceAdderSignature.
+
+    Returns:
+        None
+    """
+    parallel_sequence_adder_signature.on_building_data_server_adder_signature(
+        builder=mock_builder
+    )
+
+    assert mock_builder.store.adder_signature_fn
```

Comment:
Please rather use `hasattr`. I think you can also assert that the function is callable.

Reply:
I have made this change. I did not add a callable assertion as I call the function in the next line to test it so it would be redundant

Reply:
Thank you :)